src/Packetery/Module/Carrier/PacketaPickupPointsConfig.php

```diff
@@ -91,6 +91,31 @@ public function getCompoundCarriers(): array {
 		return $indexedCollection;
 	}
 
+	/**
+	 * Gets vendor groups for compound carrier.
+	 *
+	 * @param string $carrierId Carrier id.
+	 * @param string $country   Country.
+	 *
+	 * @return array
+	 */
+	public function getCompoundCarrierVendorGroups( string $carrierId, string $country ): array {
+		$vendorGroups              = [];
+		$compoundCarrierCollection = $this->compoundCarrierFactory->create();
+		foreach ( $compoundCarrierCollection as $compoundProvider ) {
+			if ( $compoundProvider->getId() === $carrierId ) {
+				$vendorCodes = $compoundProvider->getVendorCodes();
+				foreach ( $vendorCodes as $vendorCode ) {
+					if ( strpos( $vendorCode, $country ) === 0 ) {
```

Comment:
Viz. call - Přidat groupy rovnou ke carriers než takhle to vyzobávat přes strpos a substr a podívat se na zemi dopravce.